### PR TITLE
issue-928 WIP - timeout when reaching maxlength

### PR DESCRIPTION
**Pull request by JosiahRooney**
I think I'm going the right direction here but I experience a timeout when using the .type() method. It correctly stops typing after reaching maxlength, but then hangs until a few seconds pass and it says timeout.

### CODE REVIEW

File: packages/driver/src/cy/commands/actions/type.coffee [hunk 1]
```diff
@@ -112,6 +112,9 @@ module.exports = (Commands, Cypress, cy, state, config) ->
       if chars is ""
         $utils.throwErrByPath("type.empty_string", { onFail: options._log })
 
+      if options.$el.attr("maxlength") and chars.length > options.$el.attr("maxlength")
```
Comment (chrisbreiding) [inline packages/driver/src/cy/commands/actions/type.coffee]: `options.$el.attr("maxlength")` returns a string. We should `parseInt` the value and ignore if the result is not a number. That would consistent with Chrome's behavior, at least. `maxlength="5foo"` results in a limit of 5 characters. `maxlength="foo"` just gets ignored.

**Comment by chrisbreiding** [comment]
Can you post the exact timeout error you're encountering? I'm not able to reproduce.

**Comment by JosiahRooney** [comment]
@chrisbreiding this is the timeout I'm experiencing. It seems to be expecting the longer string as the value of the input despite truncating it.

screen shot 2017-11-20 at 11 04 33 am

I will write a test for it and submit that.

**Comment by JosiahRooney** [comment]
Oh jeez, I figured it out. I was using the wrong predication. I was using .should('contain'), when it should have been .should('have.value'). The test passes now.

File: packages/driver/src/cy/commands/actions/type.coffee [hunk 2]
```diff
@@ -112,8 +112,10 @@ module.exports = (Commands, Cypress, cy, state, config) ->
       if chars is ""
         $utils.throwErrByPath("type.empty_string", { onFail: options._log })
 
-      if options.$el.attr("maxlength") and chars.length > options.$el.attr("maxlength")
-        chars = chars.slice(0, options.$el.attr("maxlength"))
+      if options.$el.attr("maxlength")
+        maxLength = parseInt(options.$el.attr("maxlength"))
```
Comment (chrisbreiding) [inline packages/driver/src/cy/commands/actions/type.coffee]: This should use the radix param for parseInt: `parseInt(options.$el.attr("maxlength"), 10)`

File: packages/driver/test/cypress/integration/commands/actions/type_spec.coffee [hunk 3]
```diff
@@ -77,6 +77,15 @@ describe "src/cy/commands/actions/type", ->
         .then ->
           expect(blurred).to.be.true
 
+    it "limits text entered to the maxlength attribute of a text input", ->
```
Comment (chrisbreiding) [inline packages/driver/test/cypress/integration/commands/actions/type_spec.coffee]: This is great. Could you add another test covering an invalid maxlength like `maxlength="five"`?

File: packages/driver/src/cy/commands/actions/type.coffee (hunk 2 above)
Comment (JosiahRooney) [reply packages/driver/src/cy/commands/actions/type.coffee]: Oh right thanks, I often forget that it should be included.

**Comment by JosiahRooney** [comment]
This closes issue #928.

**Pull request merged**